### PR TITLE
CAE-193: add "IGNORE" options to time series commands (for v5 branch)

### PR DESCRIPTION
**Pull request by sjpotter**
### Description

> Describe your pull request here

---

### Checklist

- [ ] Does `npm test` pass with this change (including linting)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

### CODE REVIEW

File: packages/time-series/lib/commands/ADD.spec.ts [hunk 1]
```diff
@@ -57,16 +57,44 @@ describe('TS.ADD', () => {
       );
     });
 
-    it('with RETENTION, ENCODING, CHUNK_SIZE, ON_DUPLICATE, LABELS', () => {
+    it('with IGNORE no values', () => {
+      assert.deepEqual(
+        ADD.transformArguments('key', '*', 1, {
+          IGNORE: { }
```
Comment (leibale) [inline packages/time-series/lib/commands/ADD.spec.ts:63]: Is "IGNORE no values" supported? (https://github.com/RedisTimeSeries/RedisTimeSeries/pull/1543)
If it is - add support for `IGNORE: true` for this case

File: packages/time-series/lib/commands/ADD.ts [hunk 2]
```diff
@@ -11,12 +11,18 @@ import {
   Timestamp
 } from '.';
 
+export interface TsIgnoreOptions {
+  MAX_TIME_DIFF: number;
+  MAX_VAL_DIFF: number;
```
Comment (leibale) [inline packages/time-series/lib/commands/ADD.ts:16]: Is "IGNORE no values" supported? (https://github.com/RedisTimeSeries/RedisTimeSeries/pull/1543)
If it is - both `MAX_TIME_DIFF` and `MAX_VAL_DIFF` should be optional..

**Review by leibale — CHANGES_REQUESTED** [review]

File: packages/time-series/lib/commands/ADD.ts (hunk 2 above)
Comment (sjpotter) [reply packages/time-series/lib/commands/ADD.ts]: doesn't seem like it, from code

```
int parseIgnoreArgs(RedisModuleCtx *ctx,
                    RedisModuleString **argv,
                    int argc,
                    long long int *ignoreMaxTimeDiff,
                    double *ignoreMaxValDiff) {
    int idx = RMUtil_ArgIndex("IGNORE", argv, argc);
    if (idx > 0) {
        long long maxTimeDiff;
        double maxValDiff;

        if (idx + 2 >= argc ||
            RMUtil_ParseArgs(argv, argc, idx + 1, "ld", &maxTimeDiff, &maxValDiff) !=
                REDISMODULE_OK) {
            RTS_ReplyGeneralError(ctx, "TSDB: Couldn't parse IGNORE");
            return TSDB_ERROR;
        }

        if (maxTimeDiff < 0 || maxValDiff < 0) {
            RTS_ReplyGeneralError(ctx, "TSDB: IGNORE arguments cannot be negative");
            return TSDB_ERROR;
        }

        *ignoreMaxTimeDiff = maxTimeDiff;
        *ignoreMaxValDiff = maxValDiff;
    }

    return TSDB_OK;
}
```

Comment (leibale) [reply packages/time-series/lib/commands/ADD.ts]: So if not, why `MAX_TIME_DIFF` and `MAX_VAL_DIFF` are optional? also, rename them to `maxTimeDiff` and `maxValDiff` to match the command interface.

Comment (sjpotter) [reply packages/time-series/lib/commands/ADD.ts]: if we want to require the user to specify them as 0, I'm good with that as well

Comment (leibale) [reply packages/time-series/lib/commands/ADD.ts]: If the server does not have defaults we shouldn't as well..

File: packages/time-series/lib/commands/ADD.spec.ts [hunk 3]
```diff
@@ -57,14 +57,43 @@ describe('TS.ADD', () => {
       );
     });
 
+    it ('with IGNORE', () => {
+      it('no values', () => {
```
Comment (leibale) [inline packages/time-series/lib/commands/ADD.spec.ts:61]: invalid tests?

Comment (sjpotter) [reply packages/time-series/lib/commands/ADD.spec.ts:61]: yes, now, didn't remove afte removing from code above.

**Pull request merged**